engine/docs/CODE_STRUCTURE.md

```diff
@@ -0,0 +1,127 @@
+* [script](../script/README.md) - A high level scripting interface that uses Lua as the backend.
+* render - High level rendering concepts like materials, constants, cameras and render command buffer.
+* rig - Our model+skeleton animation system.
+* gameobject - Collections, game objects and component base functionality
+* gui - Our gui system. Used by `comp_gui.cpp`
+* [sound](../sound/README.md) - Our software sound mixing library
+* liveupdate - High level functionality for downloadable content
+* crash - Callstack callbacks and writing functionality
+* gamesys - The main bulk of all resource types, component types and script modules.
+* tools - Utilities like the `gdc` and the editor `launcher`
+* record - Video (.avi) recording library
+* profiler - High level profiler api (using functionality from dlib's profile.h)
+* engine - The defold engine libraries and executable
+* [sdk](../sdk/README.md) - Extra tests for our `dmSDK` headers.
+
+
+### Headers
+
+When building the engine, some headers are copied from the local library folder, to the `DYNAMO_HOME/include` sub folders.
+
+Similarly, we copy header files from any `dmsdk` sub folders from each library.
+These end up in the final SDK that our users use on a daily basis.
+
+If you edit any of those header files, be aware of any breaking changes to the api!
+
+
+### Bob.jar
+
+Our command line content build system is called `Bob.jar`.
+It is [described here](../../com.dynamo.cr/README.md)
+
+### External libraries
+
+We keep certain versions of external code in `defold/share/ext`
+Since they change so rarely, we don't want them building in our main build step.
+
+Examples are `protobuf`, `tremolo` and `luajit`
+
+The output is stored in the repository, under `defold/packages`
+
+### External libraries pt 2
+
+Since the projects under `defold/share/ext` are farily cumbersome to maintain and build, we have moved some to `defold/external`.
+
+The benefit is that we can use the same build system for all platforms, which makes it easy to maintain such a library.
+
+Currently includes `Bullet 3D` and `glfw`.
+
+The output is stored in the repository, under `defold/packages`
+
+See the [readme](../../external/README.md) for more detailed info.
+
+### Packages
+
+Ìn `defold/packages` we store code/tools that are rarely updated.
+We unpack these packages with `./build.py install_ext` and they're installed into the `DYNAMO_HOME` folder.
+
+
+## Code style
+
+### Code conventions
+
+This document isn't about the code style specifically, but it might be good to mention a few things.
+We have a detailed [blog post](https://defold.com/2020/05/31/The-Defold-engine-code-style/) about our code style, but here we list the key points.
+
+In our code we use C++ very sparingly.
+We mainly use it for namespaces, and our own container types.
+
+* No STL - We do not use STL in our code (with one single exception, and we should remove that!).
```

Comment:
What is the exception?

Reply:
That was intentionally left out to not encourage more usage of it.
But I'll just remove it :)